vizro-core/tests/unit/vizro/models/_controls/test_filter.py

```diff
@@ -111,6 +161,16 @@ def test_set_slider_values_incompatible_column_type(self, test_input, managers_o
         ):
             filter.pre_build()
 
+    def test_set_datepicker_values_incompatible_column_type(self, managers_one_page_two_graphs):
+        filter = vm.Filter(column="country", selector=vm.DatePicker())
+        model_manager["test_page"].controls = [filter]
+        with pytest.raises(
+            ValueError,
+            match=f"Chosen selector {vm.DatePicker().type} "
```

Comment:
I suppose what I meant earlier is that we could already raise this for any non datetime column to make our lives a bit easier

Reply:
Sorry, but I don't understand. Isn't that we raise and exception for ant non datetime column.

```
    # FILE: filter.py
    def _set_date_picker_values(self):
        if isinstance(self.selector, SELECTORS["temporal"]):
            if self._column_type != "temporal":
                raise ValueError(
                    f"Chosen selector {self.selector.type} is not compatible "
                    f"with {self._column_type} column '{self.column}'."
                )
           ...
```

Reply:
I was referring to this comment: https://github.com/mckinsey/vizro/pull/319#discussion_r1517664691